### DISCUSSION TITLE
Guidance or example how to pass back information to hosting Winform from hosted Blazor

### DISCUSSION THREAD

**Discussion by michaelongithub** [MicrosoftEdge/WebView2Feedback#3876] [Q&A]
Our main interest is the file system access capabilities of the hosting Winform /WPF/.....
We would like to trigger an event in the hosting Winform (that has access to it) for making file operations, like open the file for the user to edit.

How can the Blazor app call a method / trigger an event in the hosting App ! (in c#)

Would appreciate info greatly.
Michael

**Comment by champnic** [comment]
Hey @michaelongithub - just wanted to confirm, is this question about WebView2? Can you share a bit about how you are using WebView2 in your app?

**Comment by michaelongithub** [comment]
Basically we are developing Balzor apps that manage documents in the file system. We want to have the possibility to open Word, PDF etc (not download) directly in Office etc for modification. These apps are used internally by  people working in the company network, where from their notebooks or pc's have access to the necessary parts of the file system. So having a "Chaperone" Winform to host the Blazor and then telling the host to open the necessary file is the goal.
